openedx/core/djangoapps/oauth_dispatch/dot_overrides/validators.py

```diff
@@ -91,6 +91,15 @@ def save_bearer_token(self, token, request, *args, **kwargs):
         request.grant_type = grant_type
         request.user = user
 
+    def get_default_scopes(self, client_id, request, *args, **kwargs):
+        """
+        If the request payload does not have `scopes` attribute for a grant_type of
+        client credentials, it should use available scopes as default.
+        """
+        if request.grant_type == 'client_credentials' and not request.scopes:
+            return get_scopes_backend().get_available_scopes(application=request.client, request=request)
```

Comment:
Thanks. Some high-level thoughts:
1. I don't know how this will interact with the email/profile defaults implemented in the JWT code? Whatever change we might make, we need to ensure we have good test coverage.
2. See https://github.com/openedx/edx-platform/pull/33455 for related background around the user_id claim. I wish we had a cleaner simpler story for where user_id should **not** be added. It also includes default scope code, which seems to exist in other locations, so I'm not clear on the best implementation.
3. It feels like adding *all* scopes as default would make things less-secure. I'm wondering if we should keep this scoped (pun-intended) down to just `user_id`? I think we are moving to a place where we are trying to make `user_id` a default if `user_id` is an allowable scope.

Reply:
Here are some specific requests. Please let me know if you have any questions or concerns. I don't think any of this will take very long.
1. Instead of adding all allowable scopes, only add `user_id` if it is in the allowable scopes list. (Note: we can always grow the default list in the future if need be, but it will be difficult to impossible to ever shrink it.)
2. Update the ADR in https://github.com/openedx/edx-platform/pull/33455 with this addition. You can add a Change History section, like is used in OEPs.
3. Add a temporary rollout toggle setting. It is easy to get auth stuff wrong, or have unintended consequences, and this will make it simple to turn on/off. We can remove the toggle after all has proven ok.
4. Add or update unit tests. This includes tests around the JWTs. Having the rollout toggle should make it simple to test that scopes are working as we expect before and after.

@iloveagent57: You can also let me know if you think any of this seems unreasonable.

Reply:
That all sounds extremely reasonable! Thanks for reviewing this.

For my own clarity on (1) - what are the reasons to _not_ return all scopes to the requesting application when the corresponding `ApplicationAccess` record says it's ok for that application to have those scopes?